pkg/cmd/image/squash.go

```
@@ -0,0 +1,450 @@
+			return newImg, fmt.Errorf("failed to create new squashImage %s: %w", img.Name, err)
+		}
+	}
+	return newImg, nil
+}
+
+// generateCommitImageConfig returns commit oci image config based on the container's image.
+func (sr *squashRuntime) generateCommitImageConfig(ctx context.Context, baseImg images.Image, baseConfig ocispec.Image, diffID digest.Digest) (ocispec.Image, error) {
+	createdTime := time.Now()
+	arch := baseConfig.Architecture
+	if arch == "" {
+		arch = runtime.GOARCH
+		log.G(ctx).Warnf("assuming arch=%q", arch)
+	}
+	os := baseConfig.OS
+	if os == "" {
+		os = runtime.GOOS
+		log.G(ctx).Warnf("assuming os=%q", os)
+	}
+	author := strings.TrimSpace(sr.opt.Author)
+	if author == "" {
+		author = baseConfig.Author
+	}
+	comment := strings.TrimSpace(sr.opt.Message)
+
+	baseImageDigest := strings.Split(baseImg.Target.Digest.String(), ":")[1][:12]
+	return ocispec.Image{
+		Platform: ocispec.Platform{
+			Architecture: arch,
+			OS:           os,
+		},
+
+		Created: &createdTime,
+		Author:  author,
+		Config:  baseConfig.Config,
+		RootFS: ocispec.RootFS{
+			Type:    "layers",
+			DiffIDs: append(baseConfig.RootFS.DiffIDs, diffID),
+		},
+		History: append(baseConfig.History, ocispec.History{
+			Created:    &createdTime,
+			CreatedBy:  fmt.Sprintf("squash from %s", baseImageDigest),
+			Author:     author,
+			Comment:    comment,
+			EmptyLayer: false,
+		}),
+	}, nil
+}
+
+// Squash will squash the image with the given options.
+func Squash(ctx context.Context, client *containerd.Client, option types.ImageSquashOptions) error {
+	var srcName string
+	walker := &imagewalker.ImageWalker{
+		Client: client,
+		OnFound: func(ctx context.Context, found imagewalker.Found) error {
+			if srcName == "" {
+				srcName = found.Image.Name
+			}
+			return nil
+		},
+	}
+	matchCount, err := walker.Walk(ctx, option.SourceImageRef)
+	if err != nil {
+		return err
+	}
+	if matchCount < 1 {
+		return fmt.Errorf("%s: not found", option.SourceImageRef)
+	}
```

what's the behavior if `matchCount > 1`?

---

I'm not sure. Maybe just return an error.